### PR TITLE
updates to graph_notebook_config for Amazon Neptune Proxy Connection

### PR DESCRIPTION
**Pull request by cubeddu**
Issue #399 

Description of changes:

I have enhanced the documentation for the graph_notebook_config table to include all possible properties and clarify when the proxy_host and proxy_port are exclusively utilized for Amazon Neptune, plus fixed all markdown warnings in README.

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -242,15 +265,36 @@ Change the configuration using `%%graph_notebook_config` and modify the defaults
...
+  "ssl": true,
+  "proxy_port": 8182,
+  "proxy_host": "host.proxy.com",
+  "auth_mode": "IAM",
+  "aws_region": "us-east-1",
+  "load_from_s3_arn": ""
+}
+```
+
+Connecting to Amazon Neptune from clients outside the Neptune VPC using AWS Network [Load Balenacer](https://aws-samples.github.io/aws-dbs-refarch-graph/src/connecting-using-a-load-balancer/#connecting-to-amazon-neptune-from-clients-outside-the-neptune-vpc-using-aws-network-load-balancer)
```
Comment (alexey-temnikov) [inline README.md:291]:
```suggestion
Connecting to Amazon Neptune from clients outside the Neptune VPC using AWS Network [Load Balancer](https://aws-samples.github.io/aws-dbs-refarch-graph/src/connecting-using-a-load-balancer/#connecting-to-amazon-neptune-from-clients-outside-the-neptune-vpc-using-aws-network-load-balancer)
```

Comment (cubeddu) [reply README.md]: Corrected.

File: README.md [hunk 2]
```diff
@@ -170,11 +176,28 @@ After the above command completes, rerun the commands given at [Jupyter Classic
...
+| aws_region | the AWS region to use for Amazon Neptune connections | your-region-1 | string |
+| host | the host url to form a connection with | localhost | string |
+| load_from_s3_arn | the ARN of the S3 bucket to load data from [Amazon Neptune only] | | string |
+| port | the port to use when creating a connection | 8182 | number |
+| proxy_host | the proxy host url to route a connection through [Amazon Neptune only]| | string |
+| proxy_port | the proxy port to use when creating proxy connection [Amazon Neptune only] | 8182 | number |
+| ssl | whether to make connections to the created endpoint with ssl or not [True/False] | False | boolean |
+| ssl_verify | whether to verify the server's TLS certificate or not [True/False] | True | boolean |
+| sparql | SPARQL connection object | ``` { "path": "sparql" } ``` | string |
+| gremlin | Gremlin connection object | ``` { "username": "neo4j", "password": "password", "auth": true, "database": null } ```| string |
```
Comment (michaelnchin) [inline README.md:193]: The default values for Gremlin and Neo4J config sections are switched around here

**Review by michaelnchin — CHANGES_REQUESTED** [review]
Looks good overall, couple changes requested in config docs.

File: README.md [hunk 3]
```diff
@@ -170,11 +176,28 @@ After the above command completes, rerun the commands given at [Jupyter Classic
 
 ## Connecting to a graph database
 
+Configuration options can be set using the `%graph_notebook_config` magic command. The command accepts a JSON object as an argument. The JSON object can contain any of the configuration options listed below. The command can be run multiple times to change the configuration. The configuration is stored in the notebook's metadata and will be used for all subsequent queries.
+
+| Configuration Option | Description | Default Value | Type |
+| --- | --- | --- | --- |
+| auth_mode | the authentication mode to use for Amazon Neptune connections | DEFAULT | string |
```
Comment (michaelnchin) [inline README.md:183]: Capitalize the first letter in the description for each field.

File: README.md (hunk 2 above)
Comment (michaelnchin) [inline README.md:193]: The default values listed for the Gremlin and Neo4J sections are switched around here.

File: README.md (hunk 3 above)
Comment (cubeddu) [reply README.md]: Will do

File: README.md (hunk 2 above)
Comment (cubeddu) [reply README.md]: Oops, corrected that mistake.

**Comment by cubeddu** [comment]
> Looks good overall, couple changes requested in config docs.

Changes made to reflect the latest request.

**Review by michaelnchin — APPROVED** [review]
LGTM, thanks @cubeddu !

**Comment by xiazcy** [comment]
Quick followup, @michaelnchin do we still plan to merge this PR? Thanks!

**Pull request merged**